modules/public/views.py

```diff
@@ -162,24 +162,63 @@ def convert_limit_offset_to_size_page(link):
 
 
 def homepage():
+    featured_snaps = []
+    errors = []
+    try:
+        featured_snaps = normalize_searched_snaps(api.get_featured_snaps())
+    except api.InvalidResponseContent as invalid_response_content:
+        errors = ["Error from server"]
```

Comment:
It feels like we're losing information here if all we pass on is "Error from server". We know it's about the content of the response. Couldn't we be slightly more descriptive in the public message?